### PR TITLE
Add Datadog Notifier

### PR DESCRIPTION
**Pull request by caledhwa**
Adding a configurable DataDog notifier to `kit-overwatch`. This notifier requires configuration including a proper set of DataDog api key, and app key. Overwatch can be run without DataDog by leaving it disabled through `KIT_OVERWATCH_NOTIFY_DATADOG` = `false`. 

- Added files to .gitignore
- Added `generate` to make. Is an easy `make generate` command to generate code for fakes and other.
- Built simple `IDataDogClient` interface to facilitate unit testing
- Test coverage on `NotifyDataDog` at 93% (hard to test `json.Marshal` errors)
- Added docs to README, suggestions for further functionality
- Additional tag calculated: `serviceName` - calculated for easy searching in DataDog. Conditionally takes into account pod names such as `podname-service-deployment-12345-af23b-12345566.132123` and converts that to `podname-service`. Tested all scenarios.
- Chose JSON for metadata because it was the best formatting for events in DataDog. Much of the markdown rules in DataDog do not work correctly.

Events appear like this in DataDog:

screen shot 2016-11-20 at 7 22 27 am

Error events will appear highlighted in Red. All info events are written as `low` priority. Normal priority is saved for warnings and errors.

Enjoy!!! :-)

### CODE REVIEW

File: notifiers/datadog/datadog.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		event.AlertType = "Error"
+		event.Priority = "high"
+	}
+
+	serviceName := n.Event.ObjectMeta.Name
+	splitName := strings.Split(n.Event.ObjectMeta.Name, "-")
+
+	switch n.Event.InvolvedObject.Kind {
+	case "Pod":
+		// Default Rules to ignore `service-name-[deploynumber-podnumber.number]`
```
Comment (c1freitas) [inline notifiers/datadog/datadog.go]: even though this is only a comment, technically `service-name-[deploynumber-podnumber.number]` is really the `service-name-[replicaset-podnumber.number]`

**Comment by caledhwa** [comment]
@c1freitas - I appreciate your comment and updated accordingly!

File: notifiers/datadog/datadog.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+	case "ERROR":
+		event.AlertType = "Error"
+		event.Priority = "high"
+	}
+
+	serviceName := n.Event.ObjectMeta.Name
+	splitName := strings.Split(n.Event.ObjectMeta.Name, "-")
+
+	switch n.Event.InvolvedObject.Kind {
+	case "Pod":
```
Comment (chesleybrown) [inline notifiers/datadog/datadog.go]: Might be useful to break this out into a util func so we can use it in other notifiers (such as the Slack notifier).

Comment (caledhwa) [inline notifiers/datadog/datadog.go]: I totally considered that. I'll refactor it out and drop it in a `util` package. I'll stay out of the other notifiers... unless you want me to add it to log and slack. @chesleybrown

**Comment by chesleybrown** [comment]
Hm. My merge seems to have created some conflicts. Could you resolve @caledhwa and then we can get this deployed?

File: notifiers/datadog/datadog.go (hunk 2 above)
Comment (chesleybrown) [inline notifiers/datadog/datadog.go]: Yeah, lets just tackle this later. Don't want this to hold up this feature.

**Pull request merged**